packages/fxa-settings/src/pages/Signin/index.test.tsx

```diff
@@ -703,6 +679,39 @@ describe('with sessionToken', () => {
           expect(hardNavigateSpy).toHaveBeenCalledWith('someUri');
         });
       });
+      it('handles error due to TOTP required or insufficent ARC value', async () => {
+        const signinResponse = createBeginSigninResponse();
+        const beginSigninHandler = jest
+          .fn()
+          .mockReturnValueOnce(signinResponse);
+        const finishOAuthFlowHandler = jest.fn().mockImplementationOnce(() => {
+          throw AuthUiErrors.TOTP_REQUIRED;
+        });
+        const integration = createMockSigninOAuthIntegration();
+        render({ finishOAuthFlowHandler, integration, beginSigninHandler });
+
+        enterPasswordAndSubmit();
+        await waitFor(() => {
+          expect(navigate).toHaveBeenCalledWith('/inline_totp_setup', {
+            state: {
+              email: MOCK_EMAIL,
+              keyFetchToken: signinResponse.data.signIn.keyFetchToken,
+              sessionToken: signinResponse.data.signIn.sessionToken,
+              uid: signinResponse.data.signIn.uid,
+              unwrapBKey: signinResponse.data.unwrapBKey,
+              verificationMethod: signinResponse.data.signIn.verificationMethod,
+              verificationReason: signinResponse.data.signIn.verificationReason,
+              verified: signinResponse.data.signIn.verified,
+            },
+          });
+        });
+        // TODO, follow up needed for better handling...
```

Comment:
See comment further down about "A few notes for the reviewer...".

Reply:
I'm not sure what that TODO mean.

Reply:
I left a comment [in FXA-9235 ](https://mozilla-hub.atlassian.net/browse/FXA-9235?focusedCommentId=859802) referencing [this comment](https://github.com/mozilla/fxa/pull/16665#discussion_r1552554524). Basically, it looks to me like we need to wrap every `finishOAuthFlowHandler` call in a `try`/`catch` similar to this, but we also need to handle those errors. I think that's going to mean either checking for `returnOnError` or displaying something in the UI. I'll reference this ticket here next time I push.